### PR TITLE
Учебный проект «Холст»

### PR DESCRIPTION
**Pull request by costigans**
---
:mortar_board: [Учебный проект «Холст»](https://up.htmlacademy.ru/javascript/14/user/243814/tasks/10)

:boom: https://htmlacademy-javascript.github.io/243814-code-and-magick/2/

### CODE REVIEW

File: js/stat.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  ctx.lineTo(x + CLOUD_WIDTH - 20, y + CLOUD_HEIGHT - 10);
+  ctx.lineTo(x + CLOUD_WIDTH / 2, y + CLOUD_HEIGHT);
+  ctx.lineTo(x + 20, y + CLOUD_HEIGHT - 10);
+  ctx.lineTo(x + 0, y + CLOUD_HEIGHT);
+  ctx.lineTo(x + 10, y + CLOUD_HEIGHT - 20);
+  ctx.lineTo(x + 0, y + CLOUD_HEIGHT / 2);
+  ctx.lineTo(x + 10, y + 20);
+  ctx.lineTo(x + 0, y + 0);
+  ctx.closePath();
+  ctx.fill();
```
Comment (rkhaslarov) [inline js/stat.js]: Use fillRect instead of lineTo

File: js/stat.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  }
+
+  return maxElement;
+};
+
+window.renderStatistics = function (ctx, players, times) {
+  // Рисует облако с тенью
+  renderCloud(ctx, CLOUD_X + GAP, CLOUD_Y + GAP, 'rgba(0, 0, 0, 0.7)');
+  renderCloud(ctx, CLOUD_X, CLOUD_Y, '#ffffff');
```
Comment (rkhaslarov) [inline js/stat.js]: Move colors to separate constants like SHADOW_COLOR and etc

File: js/stat.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+var CLOUD_X = 100;
+var CLOUD_Y = 10;
+var CLOUD_PADDING = 30;
+var GAP = 10;
+var TEXT_HEIGHT = 20;
+var BAR_WIDTH = 40;
+var BAR_INDENT = 50;
+var barHeight = CLOUD_HEIGHT - CLOUD_PADDING * 2 - TEXT_HEIGHT * 3;
+
+var getRandomArbitrary = function (min, max) {
```
Comment (rkhaslarov) [inline js/stat.js]: rename to getRandomInRange

File: js/stat.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  ctx.fillText('Список результатов:', CLOUD_X + CLOUD_PADDING, CLOUD_Y + CLOUD_PADDING + TEXT_HEIGHT);
+
+  var maxTime = getMaxElement(times);
+
+  // Рисует результаты игроков
+  for (var i = 0; i < players.length; i++) {
+    ctx.fillText(players[i], CLOUD_X + CLOUD_PADDING + GAP + (BAR_WIDTH + BAR_INDENT) * i, CLOUD_Y + CLOUD_HEIGHT - CLOUD_PADDING + GAP);
+
+    // Вычисляет цвет столбца
+    var colorColumn = (players[i] === 'Вы') ? 'rgba(255, 0, 0, 1)' : 'rgba(0, 0, 255, ' + getRandomArbitrary(0.3, 1) + ')';
```
Comment (rkhaslarov) [inline js/stat.js]: Unnecessary variable because it used once

File: js/stat.js (hunk 2 above)
Comment (costigans) [reply js/stat.js]: fix

File: js/stat.js (hunk 3 above)
Comment (costigans) [reply js/stat.js]: fixed

File: js/stat.js (hunk 4 above)
Comment (costigans) [reply js/stat.js]: fixed

**Pull request merged**